lib/site_prism/section.rb

```diff
@@ -12,7 +12,7 @@ class Section
     def initialize(parent, root_element)
       @parent = parent
       @root_element = root_element
-      Capybara.within(@root_element) { yield(self) } if block_given?
+      within { yield(self) } if block_given?
```

Comment:
This adds an additional/unnecessary extra proc around the block execution when it gets passed to `within`.  You should modify the initializer method here to specify the block in the parameter list, and pass the block to within as `&block`, which then will yield within the `within` method.

```ruby
def initialize(parent, root_element, &block)
  # ..
  within(&block) if block_given?
end
```